### PR TITLE
unistore: wire the authz client

### PR DESCRIPTION
**Pull request by chaudyg**
**What is this feature?**

Enforce RBAC checks in unistore for folder and dashboards only (other resources are not compatible with .
This is behind the FlagAuthZGRPCServer feature toggle and works both in ST grafana and MT unistore.

Note: The compile is extremely inefficient for now, but this should be good enough to test until we implement the efficient implementation (WIP). 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/storage/unified/client.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ import (
 	"github.com/grafana/grafana/pkg/infra/tracing"
 	"github.com/grafana/grafana/pkg/services/apiserver/options"
 	"github.com/grafana/grafana/pkg/services/authn/grpcutils"
+	"github.com/grafana/grafana/pkg/services/authz"
```
Comment (ryantxu) [inline pkg/storage/unified/client.go:20]: Is this able to be 
https://github.com/grafana/authlib/blob/main/authz/client.go#L104

File: pkg/storage/unified/resource/access.go [hunk 2]
```diff
@@ -24,3 +24,67 @@ func (c *staticAuthzClient) Compile(ctx context.Context, id claims.AuthInfo, req
...
+		r, err := c.client.Check(ctx, id, authz.CheckRequest{
+			Verb:      "get",
+			Group:     req.Group,
+			Resource:  req.Resource,
+			Namespace: namespace,
+			Name:      name,
+			Folder:    folder,
+		})
+		if err != nil {
+			return false
```
Comment (IevaVasiljeva) [inline pkg/storage/unified/resource/access.go:75]: nit: I guess this is a temporary implementation, but might be helpful to have a debug log here for logging the error.

**Review by IevaVasiljeva — APPROVED** [review]
LGTM, this feels like a good first step.

Added one question and some suggestions for adding more logging.

File: pkg/storage/unified/resource/access.go [hunk 3]
```diff
@@ -24,3 +24,67 @@ func (c *staticAuthzClient) Compile(ctx context.Context, id claims.AuthInfo, req
 }
 
 var _ authz.AccessClient = &staticAuthzClient{}
+
+type resourceGroup map[string]map[string]interface{}
```
[the remark this answers is not in the document]
Comment (chaudyg) [reply pkg/storage/unified/resource/access.go:28]: should be groupResource instead, my bad .
group: dashboard.grafana.app
resource: dashboards

One group can have multiple resources.

**Pull request merged**